### PR TITLE
Fix wheel predownload

### PR DESCRIPTION
**Pull request by JanuszL**
- makes sure that perl doesn't hide env variables set
  by the test scripts during predownload package discovery process
- makes sure that setup_packages.py doesn't crash when the provided
  cuda version is not supported (there is no package for it)

## Category:
**Bug fix** (*non-breaking change which fixes an issue*)

## Description:
- makes sure that perl doesn't hide env variables set
  by the test scripts during predownload package discovery process
- makes sure that setup_packages.py doesn't crash when the provided
  cuda version is not supported (there is no package for it)

### Affected modules and functionalities:
- qa/setup_packages.py
- qa/test_template.sh

### Key points relevant for the review:
- NA

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: qa/download_pip_packages.sh [hunk 1]
```diff
@@ -10,23 +10,28 @@ do
...
+        # don't parse files that don't source test_template.sh driectly as variables
+        # are not set up properly
+        if [ -z "$(grep 'test_template.sh' $test_file)" ]; then
+            continue
+        fi
         export pip_packages=""
         echo "Checking file: " ${test_file}
         source ${test_file}
-        echo "PIPs to install: " ${pip_packages}
+        echo "Packagess to install: " ${pip_packages}
```
Comment (jantonguirao) [inline qa/download_pip_packages.sh:21]:
```suggestion
        echo "Packages to install: " ${pip_packages}
```

File: qa/download_pip_packages.sh [hunk 2]
```diff
@@ -10,23 +10,28 @@ do
     # check all test files inside
     for test_file in $(ls -f *.sh);
     do
+        # don't parse files that don't source test_template.sh driectly as variables
```
Comment (jantonguirao) [inline qa/download_pip_packages.sh:13]:
```suggestion
        # don't parse files that don't source test_template.sh directly as variables
```

File: qa/test_template.sh [hunk 3]
```diff
@@ -1,7 +1,14 @@
 #!/bin/bash
 
 set -o pipefail
-source qa/test_template_impl.sh 2>&1 | perl -pe 'use POSIX strftime;
+if [ -n "$gather_pip_packages" ]
+then
+    # perl breaks the opulation of the outise variables from the inside of the sourced
```
Comment (jantonguirao) [inline qa/test_template.sh:6]:
```suggestion
    # perl breaks the population of the outside variables from the inside of the sourced
```

**Review by jantonguirao — APPROVED** [review]

File: qa/test_template.sh (hunk 3 above)
Comment (JanuszL) [reply qa/test_template.sh:6]: Fixed

File: qa/download_pip_packages.sh (hunk 1 above)
Comment (JanuszL) [reply qa/download_pip_packages.sh:21]: Fixed

**Review by szalpal — APPROVED** [review]

**Pull request merged**